src/components/pip/pictue-in-picture.tsx

```diff
@@ -0,0 +1,155 @@
+import {withText} from 'preact-i18n';
+import {Layout, PlayerType} from '../../enums';
+import {DualScreenPlayer} from '../../types';
+import {h, Component, createRef} from 'preact';
+import {A11yWrapper} from '@playkit-js/common';
+
+const { redux,reducers , utils, style} = KalturaPlayer.ui;
+const {Tooltip, Icon} = ui.Components;
+/**
+ * mapping state to props
+ * @param {*} state - redux store state
+ * @returns {Object} - mapped state to this component
+ */
+const mapStateToProps = (state: Record<string, any>) => ({
+ isPictureInPictureSupported: state.engine.isPictureInPictureSupported,
+  isInPictureInPicture: state.engine.isInPictureInPicture,
+});
+
+const COMPONENT_NAME = 'PictureInPicture';
+
+interface PictureInPictureDualScreenProps {
+  player: KalturaPlayerTypes.Player | KalturaPlayerTypes.ImagePlayer;
+  dualScreenPlayers: Array<DualScreenPlayer>;
+  playerType: PlayerType;
+  isInPictureInPicture?: boolean| undefined;
+  isPictureInPictureSupported?: boolean| undefined;
+  pictureInPictureExitText?: string;
+  pictureInPictureText?: string;
+  layout?: Layout;
+}
+
+type AppState = {
+  classname: string;
+  ariaLabel:string;
+}
+
+  /**
+ * PictureInPicture component
+ *
+ * @class PictureInPicture
+ * @extends {Component}
+ */
+@redux.connect(mapStateToProps, utils.bindActions({...reducers.shell.actions, ...reducers.engine.actions}))
+@withText({
+  pictureInPictureText: 'controls.pictureInPicture',
+  pictureInPictureExitText: 'controls.pictureInPictureExit'
+})
+class PictureInPicture extends Component<PictureInPictureDualScreenProps, AppState>  {
+  /**
+   * Creates an instance of PictureInPicture.
+   * @memberof PictureInPicture
+   */
+  constructor(props: any) {
+    super(props);
+
+    const isSomePlayerInPip = this.isSomePlayerInPip()
+    const classname = [style.controlButton, 'picture-in-picture-dual-screen'].join(" ");
+    this.state = {
+      classname: isSomePlayerInPip ? classname.concat(" " + style.isInPictureInPicture || "") as string : classname,
+      ariaLabel: isSomePlayerInPip? this.props.pictureInPictureExitText || "" : this.props.pictureInPictureText || ""
+    }
+  }
+
+  componentDidUpdate() {
+    //this is to hide picture-in-picture-overly when mainPlayer go to picture in picture
+    if(this.props.playerType === PlayerType.VIDEO) {
+      const dualScreenPlayer = this.getDualScreenPlayer()
+      if(dualScreenPlayer && dualScreenPlayer.player == this.props.player) {
+        // @ts-expect-error
+        const videoPlayer:KalturaPlayerTypes.Player = this.props.player as KalturaPlayerTypes.Player;
+        videoPlayer.ui.store.dispatch?.(reducers.engine.actions.updateIsInPictureInPicture(false));
+      }
+    }
+  }
+
+  /**
+   * toggle pip
+   * @returns {void}
+   *
+   * @memberof PictureInPicture
+   */
+  togglePip = (): void => {
+    const player = this.props.player;
+    if(this.props.playerType === PlayerType.IMAGE) {
+      return
+    }
+
```

Comment:
I would recommend you do next:
`const { classname } = this.state`

then you can:
`let const currentClassName = ""`
`currentClassName = classname.replace(" " + style.isInPictureInPicture, "") as string;`
`currentClassName = classname.concat(" " + style.isInPictureInPicture || "") as string;`
and execute `this.updateParams(currentClassName, this.props.pictureInPictureExitText|| "")` only once after if/else block

that's easier for reading and avoid some code-duplication